pyo3-ffi/src/genericaliasobject.rs

```diff
@@ -0,0 +1,21 @@
+use crate::object::{PyObject, PyTypeObject, Py_TYPE};
+use crate::PyObject_TypeCheck;
+use std::os::raw::c_int;
+use std::ptr;
+
+#[inline]
+pub unsafe fn PyGenericAlias_CheckExact(op: *mut PyObject) -> c_int {
+    (Py_TYPE(op) == ptr::addr_of_mut!(Py_GenericAliasType)) as c_int
+}
+
+#[inline]
+pub unsafe fn PyGenericAlias_Check(op: *mut PyObject) -> c_int {
+    PyObject_TypeCheck(op, ptr::addr_of_mut!(Py_GenericAliasType))
+}
```

Comment:
These two functions aren't in the upstream C API and the FFI shouldn't implement them. If they're needed inside PyO3 then they should be implemented there. The FFI bindings are supposed to list exactly what is present in the headers.

Reply:
I removed it